subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/HttpElasticClient.kt

```diff
@@ -96,7 +93,16 @@ internal class HttpElasticClient(
         result
     }
 
-    private fun utcFormatter(pattern: String): DateFormat = SimpleDateFormat(pattern, Locale.getDefault()).apply {
-        timeZone = TimeZone.getTimeZone("UTC")
+    /**
+     * from javadoc:
+     *
+     * It is recommended to create separate format instances for each thread.
+     * If multiple threads access a format concurrently, it must be synchronized
+     * externally.
+     */
+    private fun utcFormatter(pattern: String) = object : ThreadLocal<DateFormat>() {
```

Comment:
ThreadLocal.withInitial { }

Reply:
This code used in ui tests env on 22 API emulators:
`Call requires API level 26 (current min is 21): java.lang.ThreadLocal#withInitial`
and it is not supported for now even with desugaring: https://developer.android.com/studio/write/java8-support-table